src/redfish_collector/routers/prometheus.py

```
@@ -175,17 +170,29 @@ async def read_all(serverAddress: IPvAnyAddress = Query(None), \
     except Exception as err:
-        logging.error("[%s] Read Inventory Configs failed: %s" %(serverAddress,err))
-        return
+        if 'PhysicalServer_Query' in componentMetrics:
+            componentMetrics['PhysicalServer_Query'].labels(str(serverAddress)).set(0)
+            metrics = generate_latest(registry)
+            # REQUEST_TIME.observe(time.time() - start_time)
+            logging.error(f"[{serverAddress}] Metric collection failed: {err}", exc_info=True)
+            rawPath = f'{REDFISH_DATA}RawData/{serverAddress}.json'
+            newPath = f'{REDFISH_DATA}NewData/{serverAddress}.json'
+            try:
+                with open(rawPath, 'w'): pass
```

Ruff (0.14.7)

Multiple statements on one line (colon)

(E701)